### PR TITLE
init: If user already exists in /etc/passwd set props again, do not skip

### LINKED ISSUE #175: [Error] wrong home path ends up in /etc/passwd when requesting custom --home during create
**Describe the bug**  
Incorrect user home path ends up in guest's /etc/passwd when using --home option during distrobox-create  
Somehow user entry in /etc/passwd inside guest is using values mentioned in host's /etc/passwd user entry

**To Reproduce**  
```
    distrobox-create --home /tmp/pathq/ --image ubuntu:latest ubuntu-studio.android.q
    distrobox-enter ubuntu-studio.android.q
```
```
    echo ${HOME}
    # this returns correct home (the one assigned with --home)

    cat /etc/passwd | grep $(whoami)
    # this shows something not right (shows host's user home path)
```

**Expected behavior**  
The path passed using --home should be used to populate (and maintain) user's entry in /etc/passwd

**Desktop**  
Host OS: Fedora 34 (x86_64)
Guest OS: Ubuntu 20.04 (docker ubuntu:latest)

**Additional context**  
The discrepancy causes visible issues. 
For example Android Studio uses user home path set in /etc/passwd to create many directories 
(.gradle, .android etc). This ends up polluting the Host's home because of the bug mentioned above.
It is expected that inside the container the home directory provided in create command will be used.

### PR DESCRIPTION
**Pull request by ennec-e**
This fixes #175 where home field in /etc/passwd diverges from value of $HOME
which then causes unexpected behaviour.
When user is pre-prepopulated due to some reason the old behaviour was to skip useradd.
So path in /etc/passwd could be different than one manually set (ex. when user asks for custom home).

Fix bug by force-setting properties with _usermod_ if user already exists.

### CODE REVIEW

**Comment by 89luca89** [comment]
Hey @ennec-e thanks for the contribution!

Just a couple of comments on this before merging, also does this fix #175 for you?

File: distrobox-init [hunk 1]
```diff
@@ -458,6 +458,16 @@ if ! grep -q "${container_user_name}" /etc/passwd; then
 		--gid "${container_user_gid}" \
 		"${container_user_name}"; then
 
+		printf "Warning: there was a problem setting up the user\n"
+	fi
+else
```
Comment (89luca89) [inline distrobox-init:463]: Let's just add a couple of comments here
this situation is presented when podman or docker already creates the user for us, so we find it already present

File: distrobox-init [hunk 2]
```diff
@@ -458,6 +458,16 @@ if ! grep -q "${container_user_name}" /etc/passwd; then
...
+	if ! usermod \
```
Comment (89luca89) [inline distrobox-init:464]: Given that we're adding usermod command, it would be nice if you add it to the list of checks in lines 194-197

**Review by 89luca89 — CHANGES_REQUESTED** [review]

**Comment by ennec-e** [comment]
> also does this fix #175 for you?

Yes. I've been using distrobox locally with this patch. It does resolve #175 for me.
I'll get on to making the changes you requested now.

**Pull request merged**